src/use-swr.ts

```diff
@@ -180,6 +180,11 @@ async function mutate<Data = any>(
       if (error) throw error
       return cache.get(key)
     })
+  } else if (shouldRevalidate) {
+    // `updaters` would trigger revalidate if `updaters.length > 0`.
+    // So clear promise of revalidate if `!updaters.length && shouldRevalidate`
+    delete CONCURRENT_PROMISES[key]
+    delete CONCURRENT_PROMISES_TS[key]
   }
```

Comment:
if hook is unmounted
```
mutate('key') // This will not clear the deduplicatedInterval

mutate("key", () => undefined) // This will clear the deduplicatedInterval
mutate("key", () => Promise.resolve(undefined)) This will also clear the deduplicatedInterval
```

https://github.com/vercel/swr/blob/f7876240094c02c9933757d3e8d568185122adfb/src/use-swr.ts#L107

Reply:
if hook is unmounted, the above three cases will not clear the deduplicatedInterval because `updaters && updaters.length` is true.

Reply:
I has an irrelevant problem. How did you embed code like this:

![image](https://user-images.githubusercontent.com/13298548/116069794-64331b00-a6be-11eb-93f2-c148af82325c.png)

Reply:
https://docs.github.com/cn/github/managing-your-work-on-github/creating-a-permanent-link-to-a-code-snippet